### ISSUE TITLE
In CiviCRM 4.7: fatal constraint violation while enabling the extension

### ISSUE THREAD

**Issue by mlutfy** [adixon/ca.civicrm.moneris#3]
I feel like the sql/install.sql statement is not necessary?

I had the following error while enabling the extension on CiviCRM 4.7.13:

```
INSERT INTO civicrm_managed (module, name, entity_type, entity_id)
VALUES ('ca.civicrm.moneris','Moneris','PaymentProcessorType',
  (SELECT id FROM civicrm_payment_processor_type where name = 'Moneris'))

[nativecode=1048 ** Column 'entity_id' cannot be null]
```

After I commented the statement out of the .sql file and re-installed the extension, the `civicrm_managed` table seemed OK:

```
| id | module                      | name                                                                   | entity_type          | entity_id | cleanup |
[...]
| 32 | ca.civicrm.moneris          | Moneris                                                                | PaymentProcessorType |        13 | NULL    |
| 33 | ca.civicrm.moneris          | Cron:Job.Monerisrecurringcontributions                                 | Job                  |        18 | NULL    |
```

**Comment by adixon** [comment]
I haven't tested this extension on 4.7, so you might be correct. There is a bit of trick dealing with the fact that a moneris record may or may not already exist as a payment processor type - the extension tries to take over that record, but it generally gives non-fatal errors on some mixes of install/uninstall/re-install.

**Comment by mlutfy** [comment]
I'll submit a patch, but before I forget: `CRM/Core/Payment/Moneris.mgd.php` :

```
<?php
// This file declares a managed database record of type "ReportTemplate".
// The record will be automatically inserted, updated, or deleted from the
// database as appropriate. For more details, see "hook_civicrm_managed" at:
// https://docs.civicrm.org/dev/en/latest/hooks/
return array (
  0 =>
  array (
    'module' => 'ca.civicrm.moneris',
    'name' => 'moneris',
    'entity' => 'PaymentProcessorType',
    'params' => array(
      'version' => 3,
      'name' => 'moneris',
      'title' => 'Moneris',
      'description' => 'Moneris',
      'class_name' => 'Payment_Moneris',
      'payment_type' => 1,
    ),
  ),
);
```

and delete the `install.sql`.

**Comment by adixon** [comment]
Thanks, and fixed with new release 1.3.
